src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -609,7 +609,7 @@ void CoreWorkerDirectTaskSubmitter::PushNormalTask(
                 };
             auto &lease_entry = worker_to_lease_entry_[addr];
             RAY_CHECK(lease_entry.lease_client);
-            lease_entry.lease_client->GetTaskFailureCause(lease_entry.task_id, callback);
```

Comment:
I think this breaks the oom killer , this is the task spec task id and is what is used by the raylet to track the failures, if we need to change it here we also need to change the task id in the raylet

https://buildkite.com/ray-project/oss-ci-build-pr/builds/6536#0184c5c2-b382-4d65-9ae4-8823d5fe69e6

Reply:
Ah yes, I think the raylet is never updated with the new task ID when tasks reuse an existing lease, right? I'll remove this change since changing that would definitely be out-of-scope.